extending-valtimo/access-control/registering-a-resource.md

```diff
@@ -0,0 +1,104 @@
+# Registering a resource
+
+Access control can be exercised over resources that have been registered. Valtimo registers a few resources out of 
+the box. These are:
+- `com.ritense.document.domain.impl.JsonSchemaDocument`
+- `com.ritense.document.domain.impl.searchfield.SearchField`
+- `com.ritense.note.domain.Note`
+- `com.ritense.valtimo.camunda.domain.CamundaTask`
+
+The authorization module allows developers to register their own resources, which allows said resources to be used when
+configuring permissions.
+
+## Registering a resource
+
+In order to register a resource for use in access control, there are two interfaces that should be implemented; the
+`AuthorizationSpecification`, and the `AuthorizationSpecificationFactory`.
+
+### Specification
+The `AuthorizationSpecification` is an abstract class built upon the Spring `Specification` interface. Generally, only
+two methods have to be overridden. The `toPredicate` method creates predicates that are used to enhance queries. For
+example, when a user looks at a list of documents, only the documents this user has access to will be shown.
+
+Secondly, the `identifierToEntity` method is used to go from an identifier to the actual entity. 
+
+In the example below, the `Audit` resource will be registered for access control. The  class extending 
+`AuthorizationSpecification` has to apply to this resource in order for the Authorization module to see it as a
+registered resource.
+
+```java
+class AuditSpecification(
+    authRequest: AuthorizationRequest<Audit>,
+    permissions: List<Permission>,
+    private val auditService: AuditService,
+    private val queryDialectHelper: QueryDialectHelper
+) : AuthorizationSpecification<Audit>(authRequest, permissions) {
+    override fun toPredicate(
+        root: Root<Audit>,
+        query: CriteriaQuery<*>,
+        criteriaBuilder: CriteriaBuilder
+    ): Predicate {
+        // Filter the permissions for the relevant ones and use those to  find the filters that are required
+        // Turn those filters into predicates
+        val groupList = query.groupList.toMutableList()
+        groupList.add(root.get<UUID>("id"))
+        query.groupBy(groupList)
+
+        val predicates = permissions.stream()
```

Comment:
Maybe we need to explain that we pretty much always need this bit to support the default conditions

Reply:
Added a comment to clarify